File: 367.valid-perfect-square.py
#
# @lc app=leetcode id=367 lang=python3
#
# [367] Valid Perfect Square
#

# @lc code=start
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        for i in range(0, num//2+2):
            if i*i == num:
                return True
            if i*i > num:
                return False
# @lc code=end

File: _367.valid-perfect-square.go
package leetcode

/*
 * @lc app=leetcode id=367 lang=golang
 *
 * [367] Valid Perfect Square
 */

// @lc code=start
func isPerfectSquare(num int) bool {
	for i := 0; i < num/2+2; i++ {
		if i*i == num {
			return true
		}
		if i*i > num {
			break
		}
	}
	return false
}

// @lc code=end
